server/enterprise.go

```diff
@@ -77,68 +78,68 @@ func (s *Server) getPRInfo(ctx context.Context, pr *model.PullRequest) (*EETrigg
-	return info, nil
-}
-
-func (s *Server) findWebappBranch(client *github.Client, ctx context.Context, serverPR *github.PullRequest) (string, string, error) {
-	serverBranch := serverPR.GetHead().GetRef()
-	destinationServerBranch := serverPR.GetBase().GetRef()
 
-	owner, webappBranch, err := s.getWebappBranchWithSameName(client, ctx, serverPR)
+	webappOwner, webappBranch, err := s.getSameBranchInRepo(ctx, pr, s.Config.EnterpriseWebappReponame)
 	if err != nil {
-		return "", "", err
+		return nil, err
 	}
-
 	if webappBranch == "" {
-		mlog.Debug("Setting base ref of server as webapp branch", mlog.Int("pr", serverPR.GetNumber()), mlog.String("ref", serverBranch))
-		return s.Config.Org, destinationServerBranch, nil
+		webappOwner = s.Config.Org
+		webappBranch = baseBranch
 	}
 
-	return owner, webappBranch, nil
+	info = &EETriggerInfo{
+		BaseBranch:   baseBranch,
+		EEBranch:     eeBranch,
+		ServerOwner:  serverOwner,
+		ServerBranch: pr.Ref,
+		WebappOwner:  webappOwner,
+		WebappBranch: webappBranch,
+	}
+	return info, nil
 }
 
-func (s *Server) getWebappBranchWithSameName(client *github.Client, ctx context.Context, serverPR *github.PullRequest) (owner string, branch string, err error) {
-	prAuthor := serverPR.GetUser().GetLogin()
-	ref := serverPR.GetHead().GetRef()
-	forkBranch, r, err := client.Repositories.GetBranch(ctx, prAuthor, s.Config.EnterpriseWebappReponame, ref)
+func (s *Server) getBranchWithSameName(ctx context.Context, remote string, repo string, ref string) (branch string, err error) {
+	clientGitHub := NewGithubClient(s.Config.GithubAccessToken)
+	ghBranch, r, err := clientGitHub.Repositories.GetBranch(ctx, remote, repo, ref)
 	if err != nil {
 		if r == nil {
-			return "", "", err
+			return "", err
 		}
-
 		if r.StatusCode != http.StatusNotFound {
-			return "", "", err
+			return "", err
 		}
+		return "", nil // do not err if branch is not found
+	}
+	if ghBranch == nil {
+		return "", fmt.Errorf("unexpected failure case")
+	}
+	return ghBranch.GetName(), nil
+}
 
-		upstreamBranch, r, err := client.Repositories.GetBranch(ctx, s.Config.Org, s.Config.EnterpriseWebappReponame, ref)
+func (s *Server) getSameBranchInRepo(ctx context.Context, serverPR *model.PullRequest, repo string) (owner string, branch string, err error) {
+	forkBranch, err := s.getBranchWithSameName(ctx, serverPR.Username, repo, serverPR.Ref)
+	if err != nil {
+		return "", "", err
+	}
+	if forkBranch == "" {
+		upstreamBranch, err := s.getBranchWithSameName(ctx, s.Config.Org, repo, serverPR.Ref)
 		if err != nil {
-			if r == nil {
-				return "", "", err
-			}
-
-			if r.StatusCode != http.StatusNotFound {
-				return "", "", err
-			}
-			return s.Config.Org, serverPR.GetBase().GetRef(), nil
+			return "", "", err
 		}
-
-		owner := s.Config.Org
-		mlog.Debug("Found upstream webapp branch", mlog.String("owner", owner), mlog.String("branch", upstreamBranch.GetName()))
-		return owner, upstreamBranch.GetName(), nil
+		if upstreamBranch == forkBranch {
```

Comment:
I think this should be removed. At this point, `forkBranch` is empty so we just have to check for empty string